testing/e2e/skr/provisioning-service-test/internal/provisioning_client.go

```diff
@@ -0,0 +1,290 @@
+package internal
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log/slog"
+	"net/http"
+	"net/url"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	interval              = time.Minute
+	provisioningTimeout   = 30 * time.Minute
+	deprovisioningTimeout = 90 * time.Minute
+	environmentType       = "kyma"
+	serviceName           = "kymaruntime"
+	accessTokenPath       = "/oauth/token"
+	environmentsPath      = "/provisioning/v1/environments"
+	bindingsPath          = "/bindings"
+)
+
+type ProvisioningConfig struct {
+	URL          string
+	ClientID     string
+	ClientSecret string
+	UAA_URL      string
+	PlanName     string
+	PlanID       string
+	User         string
+	InstanceName string
+	Region       string
+}
+
+type ProvisioningClient struct {
+	cfg         ProvisioningConfig
+	logger      *slog.Logger
+	cli         *http.Client
+	ctx         context.Context
+	accessToken string
+}
+
+func NewProvisioningClient(cfg ProvisioningConfig, logger *slog.Logger, ctx context.Context, timeoutSeconds time.Duration) *ProvisioningClient {
+	cli := &http.Client{Timeout: timeoutSeconds * time.Second}
+	return &ProvisioningClient{
+		cfg:    cfg,
+		logger: logger,
+		cli:    cli,
+		ctx:    ctx,
+	}
+}
+
+func (p *ProvisioningClient) CreateEnvironment() (CreatedEnvironmentResponse, error) {
+	requestBody := CreateEnvironmentRequest{
+		EnvironmentType: environmentType,
+		PlanName:        p.cfg.PlanName,
+		ServiceName:     serviceName,
+		User:            p.cfg.User,
+		Parameters: EnvironmentParameters{
+			Name:   p.cfg.InstanceName,
+			Region: p.cfg.Region,
+		},
+	}
+
+	requestBodyBytes, err := json.Marshal(requestBody)
+	if err != nil {
+		return CreatedEnvironmentResponse{}, fmt.Errorf("failed to marshal request body: %v", err)
+	}
+
+	resp, err := p.sendRequest(http.MethodPost, p.cfg.URL+environmentsPath, http.StatusAccepted, bytes.NewBuffer(requestBodyBytes))
+	if err != nil {
+		return CreatedEnvironmentResponse{}, err
+	}
+
+	var environment CreatedEnvironmentResponse
+	err = p.unmarshallResponse(resp, &environment)
+	if err != nil {
+		return CreatedEnvironmentResponse{}, err
+	}
+
+	return environment, nil
+}
+
+func (p *ProvisioningClient) GetEnvironment(environmentID string) (EnvironmentResponse, error) {
+	resp, err := p.sendRequest(http.MethodGet, p.cfg.URL+environmentsPath+"/"+environmentID, http.StatusOK, nil)
+	if err != nil {
+		return EnvironmentResponse{}, err
+	}
+
+	var environment EnvironmentResponse
+	err = p.unmarshallResponse(resp, &environment)
+	if err != nil {
+		return EnvironmentResponse{}, err
+	}
+
+	return environment, nil
+}
+
+func (p *ProvisioningClient) DeleteEnvironment(environmentID string) (EnvironmentResponse, error) {
+	resp, err := p.sendRequest(http.MethodDelete, p.cfg.URL+environmentsPath+"/"+environmentID, http.StatusAccepted, nil)
```

Comment:
Please consider extraction - function possibly.
I am not sure which is easier to grasp. String interpolation (sprintf) or concatenation.